### DISCUSSION TITLE
CLI - Export Ente Auth data

### DISCUSSION THREAD

**Discussion by ua741** [ente-io/ente#716] [Enhancements]
Add support for exporting the secrets from Ente Auth app.

**Comment by duckworth** [comment]
I think this would be useful for automation and scripting. This would allow workflows in terminal environments and it would facilitate a way to add integration with Raycast, Alfred and other launcher applications/productivity software. An api to fetch tokens programmatically for these purposes is also discussed here:

https://github.com/ente-io/ente/issues/454

**Comment by chkpwd** [comment]
Any update on this?

EDIT: So far I've exported the data using the Ente Auth application and created this alfred workflow [ente-totp](https://github.com/chkpwd/alfred-workflows/tree/main/ente-totp) for displaying the TOTP codes in Alfred.

@duckworth should suffice for now. 

@ua741 We would greatly appreciate first-party support for `ente auth cli` exporting. Would simplify the process a bit.

**Reply by ashutoshsaboo** [reply]
+1 on the above. Official export support for ente auth via cli would be simply fabulous. @ua741 @vishnukvmd

**Reply by ua741** [reply]
Hello, we will plan this after launching v1 version of Ente Photos app. Till then, please just upvote this feature if you are interested in it. Thanks!

**Reply by SoPat712** [reply]
@chkpwd i figured it out myself but do you wanna add a readme to that? 

something along the lines of `cd ~/Library/Application Support/Alfred/Alfred.alfredpreferences/workflows/user.workflow.xxxxxx`
and then `python main.py import secrets.txt`

and you don't mind if I might fork it and/or add some PR's?

**Reply by chkpwd** [reply]
- @SoPat712 Sorry I rushed to upload it. I'm working on adding more details. I've since uploaded the .alfredworkflow file as well. 

Let me know if you have any issues.

**Reply by SoPat712** [reply]
@chkpwd 

None so far, i'm just trying to get previews for the codes and times for expiration:

![image](https://github.com/user-attachments/assets/600ce7cb-03a5-4bb4-94a7-a511afe3ba28)

**Reply by chkpwd** [reply]
@SoPat712 

Mind creating an issue at upstream? We can continue the discussion there to further develop the workflow.

**Reply by SoPat712** [reply]
@chkpwd I did just that

**Comment by ua741** [comment]
Latest version of the cli supports exporting the codes.

**Reply by huyz** [reply]
Does it support querying for codes for integration into automated flows or other tools like Raycast?

**Reply by chkpwd** [reply]
While Ente CLI doesn't support querying TOTP codes based on search terms, there's a Raycast extension that integrates with [Ente CLI](https://www.raycast.com/chkpwd/ente-auth). There's an [Alfred extension](https://github.com/chkpwd/alfred-ente-auth) as well.
